### DISCUSSION TITLE
Can I export a forawrd and backward graph by onnxruntime

### DISCUSSION THREAD

**Discussion by JF-D** [microsoft/onnxruntime#8729] [Other Q&A] [answered]
Hi, onnxruntime community,
I am quite new with onnxruntime. But I have some questions about onnxruntime, and I didn't find any docs about that.
1. How do onnxruntime do autodiff? I mean, it is like pytorch, get the backward graph with gradient tape when running, or firstly build the backward graph, and then run the graph to do training.
2. Can I export the full graph (forward + backward + parameter update graph) from onnxruntime in a format like onnx? Is there any examples?

Thanks a lot!

**Comment by ytaous** [comment, marked as answer]
Hi, please see similar ask here - https://stackoverflow.com/questions/68672250/how-to-save-in-pytorch-an-onnx-model-with-training-autograd-operations
Hope this helps.

**Reply by JF-D** [reply]
Got it. Thanks a lot!
